axum/src/extract/host.rs

```diff
@@ -0,0 +1,59 @@
+use super::{FromRequest, RequestParts};
+use async_trait::async_trait;
+use std::{convert::Infallible};
+
+/// Extractor that extracts the host from a request.
+#[derive(Debug, Clone, Default)]
+pub struct Host(pub String);
+
+#[async_trait]
+impl<B> FromRequest<B> for Host
+where
+    B: Send,
+{
+    type Rejection = Infallible;
+
+    async fn from_request(req: &mut RequestParts<B>) -> Result<Self, Self::Rejection> {
+        if let Some(host) = req.uri().host() {
+            return Ok(Host(host.to_string()));
```

Comment:
Nitpick about the style: I would much prefer if all these `to_string()` calls on `str`s were `to_owned()` instead.